src/giftCards/GiftCardUpdatePage/GiftCardUpdateDetailsCard/TimePeriodField.tsx

```diff
@@ -0,0 +1,53 @@
+import TextWithSelectField from "@saleor/components/TextWithSelectField";
+import { FormChange } from "@saleor/hooks/useForm";
+import { TimePeriodTypeEnum } from "@saleor/types/globalTypes";
+import React from "react";
+import { useIntl } from "react-intl";
+
+import { timePeriodTextWithSelectFieldMessages as messages } from "./messages";
+
+interface TimePeriodFieldProps<T> {
+  change: FormChange;
+  periodAmount: number;
+  periodType: TimePeriodTypeEnum;
+  amountFieldName: keyof T;
+  typeFieldName: keyof T;
```

Comment:
What's the use of generic typ here? What is T, you never define the original object. `string` seems more fitting